compiler-plugin/src/main/java/io/ballerina/stdlib/http/compiler/HttpCompilerPluginUtil.java

```diff
@@ -209,10 +207,16 @@ public static TypeDescKind retrieveEffectiveTypeDesc(TypeSymbol descriptor) {
         return typeDescKind;
     }
 
-    private static boolean isBasicTypeDesc(TypeDescKind kind) {
-        return kind == TypeDescKind.STRING || kind == TypeDescKind.INT || kind == TypeDescKind.FLOAT ||
-                kind == TypeDescKind.DECIMAL || kind == TypeDescKind.BOOLEAN || kind == TypeDescKind.JSON ||
-                kind == TypeDescKind.XML || kind == TypeDescKind.RECORD || kind == TypeDescKind.BYTE;
+    public static boolean isAnyDataType(TypeDescKind kind) {
+        return kind == TypeDescKind.BOOLEAN || kind == TypeDescKind.INT || kind == TypeDescKind.FLOAT ||
+                kind == TypeDescKind.DECIMAL || kind == TypeDescKind.STRING || kind == TypeDescKind.XML ||
+                kind == TypeDescKind.JSON || kind == TypeDescKind.RECORD || kind == TypeDescKind.ANYDATA ||
+                kind == TypeDescKind.NIL || kind == TypeDescKind.BYTE || kind == TypeDescKind.STRING_CHAR ||
+                kind == TypeDescKind.XML_ELEMENT || kind == TypeDescKind.XML_COMMENT ||
```

Comment:
I don't think having all the inbuilt sub-types is a good approach, since we also allow user to create subtypes from basic types. Consider the following example : 
```ballerina
import ballerina/http;

type newXml xml;

service on new http:Listener(9090) {
    resource function get .() returns newXml {
        return xml`<name>Tharmigan</name>`;
    }
}
```

I think the above example will fail with this change since the `newXml` is not considered in the subtypes. Ideally, these type kinds should fall under `TYPE_REFERENCE` and we should allow this type if its effective type is allowable basic type. WDYT?

Reply:
I'll add a test case to cover above case

Reply:
@TharmiganK As you said, the custom subtypes which created by the user are resolved to their basic effective type. In this PR, the built-in subtypes are added which we had missed earlier.

```
type newXml xml;
```
Above is resolved to the basic type of `xml`, but
```
```
type newXmlElement xml:Element;
```
is not resolved to the basic type of `xml`. It resolved to the builtin subtype of `xml:Element`.

Hence we need to add subtype also to the condition.

Reply:
Yes, that is correct, but `xml:Element` is also a type of `xml` right? So even without adding `xml:Element` to the condition I think this will get resolve in two steps:
1. Will get the type as `newXmlElement` -> go to `TYPE_REFERENCE` case -> get the effective type as `xml:Element` -> will call `validateReturnType`
2. Will get the type as `xml:Element` -> go to `TYPE_REFERENCE` case (when the subtype is not added in the condition` -> get the effective type as `xml` -> will call `validateReturnType`
Finally now we have our basic type : `xml`
Am I correct 🤔

Reply:
In additon to the compiler tests, can we also add a ballerina test case to make sure the `PayloadBuilder` works with custom types

Reply:
> Yes, that is correct, but `xml:Element` is also a type of `xml` right? So even without adding `xml:Element` to the condition I think this will get resolve in two steps:
> 
> 1. Will get the type as `newXmlElement` -> go to `TYPE_REFERENCE` case -> get the effective type as `xml:Element` -> will call `validateReturnType`
> 2. Will get the type as `xml:Element` -> go to `TYPE_REFERENCE` case (when the subtype is not added in the condition -> get the effective type as `xml` -> will call `validateReturnType`
>    Finally now we have our basic type : `xml`
> 
> Am I correct 🤔
> 
> If so we don't need to add the subtypes to the condition

No, As I mentioned above, `xml:Element` is not further resolved to `xml`.

Reply:
Got it 👍